components/style/values/specified/position.rs

```diff
@@ -17,39 +17,104 @@ use values::specified::{LengthOrPercentage, Percentage};
 #[derive(Debug, Clone, PartialEq, Copy)]
 #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
 pub struct Position {
-    pub horizontal: LengthOrPercentage,
-    pub vertical: LengthOrPercentage,
+    pub horiz_keyword: Option<Keyword>,
+    pub horiz_position: Option<LengthOrPercentage>,
+    pub vert_keyword: Option<Keyword>,
+    pub vert_position: Option<LengthOrPercentage>,
 }
 
 impl ToCss for Position {
     fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-        try!(self.horizontal.to_css(dest));
+        // TODO: canaltinova: We should add keywords, probably?
```

Comment:
Yes, this could also be just a keyword. Print the components in order, with spaces between them if they exist.